pkg/controller/replication/replication_controller_test.go

```diff
@@ -1355,7 +1362,54 @@ func TestReadyReplicas(t *testing.T) {
 	manager.syncReplicationController(getKey(rc, t))
 
 	// ReadyReplicas should go from 0 to 2.
-	rc.Status = api.ReplicationControllerStatus{Replicas: 2, ReadyReplicas: 2, ObservedGeneration: 1}
+	rc.Status = api.ReplicationControllerStatus{Replicas: 2, ReadyReplicas: 2, AvailableReplicas: 2, ObservedGeneration: 1}
+
+	decRc := runtime.EncodeOrDie(testapi.Default.Codec(), rc)
+	fakeHandler.ValidateRequest(t, testapi.Default.ResourcePath(replicationControllerResourceName(), rc.Namespace, rc.Name)+"/status", "PUT", &decRc)
+	validateSyncReplication(t, &fakePodControl, 0, 0, 0)
+}
+
+func TestAvailableReplicas(t *testing.T) {
+	// This is a happy server just to record the PUT request we expect for status.Replicas
+	fakeHandler := utiltesting.FakeHandler{
+		StatusCode:   200,
+		ResponseBody: "{}",
+	}
+	testServer := httptest.NewServer(&fakeHandler)
+	defer testServer.Close()
+
+	c := clientset.NewForConfigOrDie(&restclient.Config{Host: testServer.URL, ContentConfig: restclient.ContentConfig{GroupVersion: testapi.Default.GroupVersion()}})
+	manager := NewReplicationManagerFromClient(c, controller.NoResyncPeriodFunc, BurstReplicas, 0)
+	manager.podStoreSynced = alwaysReady
+
+	// Status.Replica should update to match number of pods in system, 1 new pod should be created.
+	rc := newReplicationController(2)
+	rc.Status = api.ReplicationControllerStatus{Replicas: 2, ReadyReplicas: 0, ObservedGeneration: 1}
+	rc.Generation = 1
+	rc.Spec.MinReadySeconds = 20
+	manager.rcStore.Indexer.Add(rc)
+
+	// First pod becomes ready at 00:00
+	moment := unversioned.Date(2016, 11, 13, 0, 0, 0, 0, time.UTC)
+	pod := newPod("pod", rc, api.PodRunning, &moment)
+	manager.podStore.Indexer.Add(pod)
+	// Second pod becomes ready at 00:10
+	otherMoment := unversioned.Date(2016, 11, 13, 0, 0, 10, 0, time.UTC)
```

Comment:
Not a fan of using time.Now() in tests but I went ahead and made this change, ptal